neuralforecast/core.py

```diff
@@ -422,6 +451,99 @@ def cross_validation(
         fcsts_df = fcsts_df.merge(df, how="left", on=["unique_id", "ds"])
         return fcsts_df
 
+    def predict_insample(self, step_size: int = 1):
+        """Predict insample with core.NeuralForecast.
+
+        `core.NeuralForecast`'s `predict_insample` uses stored fitted `models`
+        to predict historic values of a time series from the stored dataframe.
+
+        Parameters
+        ----------
+        step_size : int (default=1)
+            Step size between each window.
+
+        Returns
+        -------
+        fcsts_df : pandas.DataFrame
+            DataFrame with insample predictions for all fitted `models`.
+        """
+        if not self._fitted:
+            raise Exception(
+                "The models must be fitted first with `fit` or `cross_validation`."
+            )
+
+        for model in self.models:
+            if model.SAMPLING_TYPE == "recurrent":
+                warnings.warn(
+                    f"Predict insample might not provide accurate predictions for \
+                       recurrent model {repr(model)} yet due to scaling."
+                )
+                print(
+                    f"WARNING: Predict insample might not provide accurate predictions for \
+                      recurrent model {repr(model)} yet due to scaling."
+                )
+
+        cols = []
+        count_names = {"model": 0}
+        for model in self.models:
+            model_name = repr(model)
+            count_names[model_name] = count_names.get(model_name, -1) + 1
+            if count_names[model_name] > 0:
+                model_name += str(count_names[model_name])
+            cols += [model_name + n for n in model.loss.output_names]
+
+        # Remove test set from dataset and last dates
+        test_size = self.models[0].get_test_size()
+        if test_size > 0:
+            dataset = TimeSeriesDataset.trim_dataset(
```

Comment:
Can we rename `dataset` to `trimmed_dataset`?
That way is easier to follow the intuition of the code.